src/partial-result-stream.ts

```diff
@@ -187,20 +196,57 @@ export class PartialResultStream extends Transform implements ResultEvents {
         .fields as google.spanner.v1.StructType.Field[];
     }
 
+    let res = true;
     if (!is.empty(chunk.values)) {
-      this._addChunk(chunk);
+      res = this._addChunk(chunk);
     }
 
-    next();
+    if (res || !this._requestStream) {
```

Comment:
In what scenario will there not be a `_requestStream` defined?

Reply:
Good point, that was a left-over from a previous version of the code where it could have been missing, but that is no longer possible.